spec/factories/courses.rb

```diff
@@ -125,6 +125,11 @@
       # We've just created a course with this provider's code, so ensure it's
       # up-to-date and has this course loaded.
       course.provider.reload
+
+      # Make sure this Course and and Sites are associated with the same Provider
+      if course.site_statuses.any?
+        course.site_statuses.first.site.provider = course.provider
```

Comment:
Hrm. A side-effect of doing it this way, I think, is that if a test tries to associate a `site` with a `course` that is from another `provider`, say to test a validation or something similar, then this is going to over-write that.

The original comment I left was about making sure that the site created by default had the correct association. I think we should be a bit more cautious about how we do that.